app/ui-utils/client/lib/openRoom.js

```diff
@@ -61,34 +54,23 @@ export const openRoom = async function(type, name) {
 				return FlowRouter.go('direct', { rid: room._id }, FlowRouter.current().queryParams);
 			}
 
+
 			if (room._id === Session.get('openedRoom') && !FlowRouter.getQueryParam('msg')) {
 				return;
 			}
 
-			if (RoomManager.open(type + name).ready() !== true) {
-				if (settings.get('Accounts_AllowAnonymousRead')) {
-					appLayout.render('main');
-				}
+			RoomManager.open(type + name);
```

Comment:
New or not?

Reply:
this is an old method, responsible for fetch the messages, prepare the subscriptions and few other actions. its 100% necessary still